src/core/server/http/http_service.ts

```diff
@@ -94,6 +98,23 @@ export class HttpService implements CoreService<HttpServiceSetup, HttpServiceSta
     };
   }
 
+  public async createServer(config: HttpConfig) {
+    const port = config.port;
+
+    if (!port || this.secondaryServers.has(port)) {
+      throw new Error(`port ${port} is already in use`);
+    }
+
+    const baseConfig = await this.config$.pipe(first()).toPromise();
+    const cfg = Object.assign({}, baseConfig, config);
```

Comment:
This can be written with spread syntax:

```js
const cfg = { ...baseConfig, ...config };
```

Reply:
In general should we prefer the spread operator over the `Object.assign` method?

Reply:
Yes, if it makes sense such as creating shallow copies or avoiding mutation. If you need to mutate an object in-place, `Object.assign`.